### PR TITLE
[WIP] Wdb 81/setup forgot password

### PR DESCRIPTION
**Pull request by kibartas**
- Implemented Forgot Password Card + Forgot Password Page
- Forgot Password Card makes a call to backend on button "Send" click

### CODE REVIEW

File: frontend/src/containers/LoginPage/LoginPage.js [hunk 1]
```diff
@@ -19,10 +20,10 @@ class LoginPage extends React.Component {
     authenticate(mail, password)
       .then((responseToken) => {
         const token = responseToken.data;
-        getCurrentUser(token)
+        window.sessionStorage.setItem('token', token);
```
Comment (AArlauskas) [inline frontend/src/containers/LoginPage/LoginPage.js:23]: I don't think moving this is a good idea. If the first request goes through and the second one does not (unlikely scenario) user would have a JWT token that would redirect him to Library page. However, we would not have all of his information. I think that we should save things to sessionStorage only when all of the requests go through successfully.

File: frontend/src/containers/ForgotPasswordPage/ForgotPasswordPage.js [hunk 2]
```diff
@@ -1,8 +1,65 @@
...
+    this.state = {
+      showServerError: false,
+      showSuccessMessage: false,
+    };
+  }
+
+  handleSendClick = async (email) => {
+    try {
+      const response = await sendForgotPasswordEmail(email);
+      if (response.status !== 200) {
```
Comment (AArlauskas) [inline frontend/src/containers/ForgotPasswordPage/ForgotPasswordPage.js]: We are already using ".then" and ".catch". Maybe we should stick with those or agree on a standard?

**Review by AArlauskas — APPROVED** [review]
Overall, looks good. Added a few comments to look into. Hoping to see backend implementation soon 😉

File: frontend/src/api/UserAPI.js [hunk 3]
```diff
@@ -6,10 +6,17 @@ const Api = axios.create({
...
     'Content-Type': 'application/json',
-    Authorization: `Bearer ${window.sessionStorage.getItem('token')}`,
   },
 });
 
+// eslint-disable-next-line
+export const getCurrentUser = () =>
+  Api.get('/users/current', {
+    headers: {
+      Authorization: `Bearer ${sessionStorage.getItem('token')}`,
```
Comment (tmaciulis22) [inline frontend/src/api/UserAPI.js:16]: Why it was moved here?

**Review by tmaciulis22 — APPROVED** [review]
🔍

File: frontend/src/api/UserAPI.js (hunk 3 above)
Comment (kibartas) [reply frontend/src/api/UserAPI.js:16]: It's folly to initialize Axios with a token that may or may not be there. The token should be added to each request individually so it's fetched when the request is called

**Pull request merged**